Week3/homework/step2-3.js

```diff
@@ -30,8 +45,17 @@ console.log('while', repeatStringNumTimesWithWhile('abc', 3));
 function repeatStringNumTimesWithDoWhile(str, num) {
   // eslint-disable-next-line prefer-const
   let result = '';
-
-  // Replace this comment and the next line with your code
+  const repeatArr = [];
+  if (num <= 0) {
```

Comment:
The `do-while` loop, on the other hand, checks the loop condition only at the end of the loop. Therefore, without an extra check it will execute the loop body at least once. However, there is no need to re-initialize `result` with an empty string as it already has that value:

```js
let result = '';
if (num > 0) {
  const repeatArr = [];
  let i = 0;
  while (i < num) {
    repeatArr.push(str);
    i++;
  }
  result = repeatArr.join('');
}
```

Reply:
The change that you made boils down to letting the error happen and then fixing it afterwards. The correct way of handling this is to not let the error occur at all. This is still something I would like you to fix.

Reply:
I just used '  let result;  '  now.

Reply:
That is not quite what I meant. You should not execute the do-while loop at all if the loop condition `num > 0` does not hold true. Otherwise you might fall off the cliff before you know it, just like the rabbit here:

https://github.com/HackYourFuture/fundamentals/blob/master/fundamentals/loops.md#dowhile

Reply:
oh..! yes understand. 
I used an if statement. I make sense.
is ti right?